tests/integrations/raftstore/test_conf_change.rs

```diff
@@ -576,12 +576,26 @@ fn test_transfer_leader_safe<T: Simulator>(cluster: &mut Cluster<T>) {
     // Test adding nodes.
     pd_client.must_add_peer(region_id, new_peer(2, 2));
     pd_client.must_add_peer(region_id, new_peer(3, 3));
-    cluster.transfer_leader(region_id, new_peer(3, 3));
-    cluster.reset_leader_of_region(region_id);
-    assert_ne!(cluster.leader_of_region(region_id).unwrap().get_id(), 3);
+    // transfer to all followers
+    let mut leader_id = cluster.leader_of_region(region_id).unwrap().get_id();
+    for peer in cluster.get_region(b"").get_peers() {
+        if peer.get_id() == leader_id {
+            continue;
+        }
+        cluster.transfer_leader(region_id, peer.clone());
+        cluster.reset_leader_of_region(region_id);
```

Comment:
I don't think the test will fail without your changes.

Reply:
That ensures we every time the check the leader is not from the cache. I think it makes the tests not always be a success.